src/tax/capitalGains.test.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+
+import moment from "moment";
+import type {GainsRecord} from "./capitalGains";
+import {CapitalGainsCalculator} from "./capitalGains";
+
+describe("capital gains calculation", () => {
+  const date = (y, m, d) =>
+    moment().set({
+      year: y,
+      month: m,
+      date: d,
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+
+  describe("short-term vs long-term: precise time boundaries", () => {
+    function expectTerm(acquireDate, disposeDate, gainsType) {
+      const calc = new CapitalGainsCalculator("foo");
+      calc.acquire(acquireDate, 1, 1);
+      calc.dispose(disposeDate, 1, 1);
+      expect(calc.gains[0].type).toEqual(gainsType);
+    }
+    const acquireDate = date(2015, 0, 0);
```

Comment:
More shared state.

Reply:
Moment really should be immutable. Fixed.